TakingITGlobal/commit2act
# [TIG-696] Data hook for countries and provinces

## Ticket

* [TIG-696](https://app.clickup.com/t/9009201449/TIG-696)

## Description

This PR introduces the `useSelectCountry` hook which [pulls a full list of countries and country codes](https://www.npmjs.com/package/i18n-iso-countries) matching the [ISO 3166-1 standard](https://en.wikipedia.org/wiki/ISO_3166-1#Officially_assigned_code_elements), as well as i18n translations for each country name in FR and ES. We're also returning a list of Canadian provinces and territories with translations matching those used in [English](https://www.canada.ca/en/immigration-refugees-citizenship/services/new-immigrants/prepare-life-canada/provinces-territories.html) and [French](https://www.canada.ca/fr/immigration-refugies-citoyennete/services/nouveaux-immigrants/preparer-vie-canada/provinces-territoires.html) on the Government of Canada website. 

I haven't been able to locate official-level Spanish translations for Canadian provinces at this time, so the hook returns the English names when a match isn't found for the user's current language.

Finally, the hook returns a `getProvincesForCountry` utility which will return the list of provinces if it is passed Canada's country code (`CA`) and an empty array for any other country. This can be used to help render the conditional "Province" selector when implemented on the `InfoSecondary` page.

## Testing

1. Pull down and install the branch (`npm install`)
2. Run all tests (`npm run test`) and confirm they pass
3. Run the app locally (`npm run start`) and visit `/account/create/details`
4. With the console open, confirm that countries and provinces are both logged
5. Repeat the same check with Storybook by visiting Pages > Sign Up > Step 2

## Screenshots

**Countries and provinces logged in browser console on the `InfoSecondary` page:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/08074680-892b-4b09-9c60-4fc933c8ff1e)
